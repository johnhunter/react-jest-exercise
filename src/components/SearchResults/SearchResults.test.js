import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import SearchResults from './';

Enzyme.configure({ adapter: new Adapter() });

const mockArticles = [{
  webUrl: 'http://google.com',
  webTitle: 'This is Google'
}];

describe('SearchResults component', ()=>{

  it('renders', ()=>{
    const wrapper = shallow(
      <SearchResults
        articles={mockArticles}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders empty when no data', ()=>{
    const wrapper = shallow(<SearchResults/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('doesnt break when no articles', ()=>{
    const wrapper = shallow(<SearchResults/>);
    expect(wrapper.find('li')).toHaveLength(0);
  });

  it('doesnt break when empty article list', ()=>{
    const wrapper = shallow(<SearchResults articles={[]}/>);
    expect(wrapper.find('li')).toHaveLength(0);
  });
});
