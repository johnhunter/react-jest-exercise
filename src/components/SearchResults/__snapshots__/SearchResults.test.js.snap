// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component renders 1`] = `
<ul>
  <li
    key="http://google.com"
  >
    <a
      href="http://google.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      This is Google
    </a>
  </li>
</ul>
`;

exports[`SearchResults component renders empty when no data 1`] = `<ul />`;
